htdocs/product/stock/card.php

```diff
@@ -620,15 +630,24 @@
 						print "</a></td>";
 					}
 
-					print "</tr>";
-					$i++;
-				}
-				$db->free($resql);
+                    if ($user->rights->stock->creer) {
```

Comment:
It seems that the test on "if ($user->rights->stock->creer)" is already present just before. This if become duplicate after patch.

Reply:
This part of code is still duplicated (the if and its content is done twice)

Reply:
I hope to have succeeded in it
I also fixed a condition test and added display units type if products in warehouse are of same nature